pkg/apis/serving/v1beta1/component.go

```diff
@@ -269,6 +278,9 @@ type ExternalMetrics struct {
 	// For namespaced query
 	// +optional
 	Namespace string `json:"namespace,omitempty"`
+	// authModes defines the authentication modes for the metrics backend
+	// +optional
+	AuthModes string `json:"authModes,omitempty"`
```

Comment:
document the possible authModes? if it is TLS additional metadata are needed, see https://keda.sh/docs/2.17/scalers/prometheus/#authentication-parameters

Reply:
also this needs be defined along side with AuthenticationRef ?